src/fsharp/ConstraintSolver.fs

```diff
@@ -2028,7 +2028,7 @@ and ReportNoCandidatesError (csenv:ConstraintSolverEnv) (nUnnamedCallerArgs,nNam
                     let errid =
                         let suggestNamesForMissingArguments = 
                             if nReqd > nActual then 
-                                let missingArgs = List.drop nReqd cmeth.AllUnnamedCalledArgs
+                                let missingArgs = List.skip nReqd cmeth.AllUnnamedCalledArgs
```

Comment:
``List.drop`` is not an exact replacement for ``List.skip`` because it doesn't throw an exception when the list is too short.  I think it's best to revert this to be on the safe side.